tests/console/firewall_enabled.pm

```diff
@@ -21,10 +21,15 @@ use warnings;
 use testapi;
 use version_utils "is_upgrade";
 use Utils::Architectures 'is_ppc64le';
+use service_check;
 
 sub run {
     my ($self) = @_;
-    if ($self->firewall eq 'firewalld') {
+    my $firewall = $self->firewall;
+    # We need to check firewalld not SuSEfirewall2, if we have done
+    # SuSEfirewall2 to firewalld service check.
+    $firewall = 'firewalld' if (($firewall eq 'SuSEfirewall2') && (exists $default_services->{susefirewall}->{service_check_func}));
```

Comment:
I suggest that the test module "firewall_enabled" does what is expected when one reads the name, i.e. just check that the firewall is enabled, not change the expectations on the fly. Can you move this code to the module that changes the firewall from SuSEfirewall2 to firewalld?

Reply:
@okurz  Hi Oliver, I've changed the code for your comments and please take a look again.

Reply:
@okurz  Hi Oliver, I've changed the code for your comments and please take a look again.